content/en/platform/corda/4.4/open-source/upgrading-cordapps.md

```diff
@@ -342,7 +342,7 @@ There are two types of contract/state upgrade:
 * *Explicit:* By creating a special *contract upgrade transaction* and getting all participants of a state to sign it using the
 contract upgrade flows.
 
-The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](api-contract-constraints.md#implicit-vs-explicit-upgrades).
+The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](api-contract-constraints.html#implicit-vs-explicit-upgrades).
 
```

Comment:
There is another broken link on this page. There's a link to "flow drains" that is missing the section link "#flow-drains"